community/modules/scheduler/htcondor-central-manager/main.tf

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  # This label allows for billing report tracking based on module.
+  labels = merge(var.labels, { ghpc_module = "htcondor-central-manager" })
+}
+
+locals {
+  network_storage_metadata = var.network_storage == null ? {} : { network_storage = jsonencode(var.network_storage) }
+  oslogin_api_values = {
+    "DISABLE" = "FALSE"
+    "ENABLE"  = "TRUE"
+  }
+  enable_oslogin_metadata = var.enable_oslogin == "INHERIT" ? {} : { enable-oslogin = lookup(local.oslogin_api_values, var.enable_oslogin, "") }
+  metadata                = merge(local.network_storage_metadata, local.enable_oslogin_metadata, var.metadata)
+
+  name_prefix = "${var.deployment_name}-cm"
+
+  all_runners = flatten([var.central_manager_runner, local.schedd_runner])
+
+  cm_config = templatefile("${path.module}/templates/condor_config.tftpl", {})
+
+  cm_object = "gs://${var.htcondor_bucket_name}/${google_storage_bucket_object.cm_config.output_name}"
+  schedd_runner = {
+    type        = "ansible-local"
+    content     = file("${path.module}/files/htcondor_configure.yml")
+    destination = "htcondor_configure.yml"
+    args = join(" ", [
+      "-e config_object=${local.cm_object}",
+    ])
+  }
+
+  central_manager_ips  = [data.google_compute_instance.cm.network_interface[0].network_ip]
+  central_manager_name = data.google_compute_instance.cm.name
+
+  list_instances_command = "gcloud compute instance-groups list-instances ${data.google_compute_region_instance_group.cm.name} --region ${var.region} --project ${var.project_id}"
+}
+
+data "google_compute_image" "htcondor" {
+  family  = var.instance_image.family
+  project = var.instance_image.project
+
+  lifecycle {
+    postcondition {
+      condition     = self.disk_size_gb <= var.disk_size_gb
+      error_message = "var.disk_size_gb must be set to at least the size of the image (${self.disk_size_gb})"
+    }
+  }
+}
+
+data "google_compute_zones" "available" {
+  project = var.project_id
+  region  = var.region
```

Comment:
It seems like there is no option to pick a zone or turn off HA. Does this pose a potential issue for egress?

Reply:
I will also create a task to add var.zones to the other 2 infrastructure modules of the HTCondor solution.

Reply:
LGTM